src/bin/server/attestations/mod.rs

```diff
@@ -0,0 +1,557 @@
+            aggregate_public_key
+        FROM
+            aggregate_interval_attestations
+        WHERE
+            interval_size = ?1
+        AND
+            slot_number = ?2
+        AND
+            value = ?3;
+        ",
+            interval_size,
+            slot_number,
+            value,
+        )
+        .fetch_optional(db_pool)
+        .await?
+    {
+        (
+            entry.num_validators + 1,
+            AggregateSignature::deserialize(&hex::decode(entry.aggregate_signature)?)
+                .map_err(|_| eyre::eyre!("Invalid aggregate signature in DB"))?,
+            AggregatePublicKey::aggregate(&[
+                PublicKey::deserialize(&hex::decode(entry.aggregate_public_key)?)
+                    .map_err(|_| eyre::eyre!("Invalid aggregate public key in DB"))?,
+                validator_public_key.clone(),
+            ])
+            .map_err(|_| eyre::eyre!("Invalid aggregate public key in DB"))?,
+        )
+    } else {
+        (
+            1,
+            AggregateSignature::infinity(),
+            AggregatePublicKey::aggregate(&[validator_public_key.clone()])
+                .map_err(|_| eyre::eyre!("Invalid aggregate public key"))?,
+        )
+    };
+
+    aggregate_signature.add_assign(&message.signature);
+    let new_aggregate_signature = hex::encode(aggregate_signature.serialize());
+    let new_aggregate_public_key = hex::encode(aggregate_public_key.to_public_key().serialize());
+
+    if new_num_validators == 1 {
+        // Create new db entry
+        sqlx::query!(
+            "
+            INSERT INTO aggregate_interval_attestations(
+                value,
+                interval_size,
+                slot_number,
+                num_validators,
+                aggregate_signature,
+                aggregate_public_key
+            )
+            VALUES (
+                ?1,
+                ?2,
+                ?3,
+                ?4,
+                ?5,
+                ?6
+            );
+            ",
+            value,
+            interval_size,
+            slot_number,
+            new_num_validators,
+            new_aggregate_signature,
+            new_aggregate_public_key,
+        )
+        .execute(db_pool)
+        .await?;
+    } else {
+        // Update existing db entry
+        sqlx::query!(
+            "
+            UPDATE aggregate_interval_attestations
+            SET
+                num_validators = ?1,
+                aggregate_signature = ?2
+            WHERE
+                interval_size = ?3
+            AND
+                slot_number = ?4
+            AND
+                value = ?5;
+            ",
+            new_num_validators,
+            new_aggregate_signature,
+            interval_size,
+            slot_number,
+            value,
+        )
+        .execute(db_pool)
+        .await?;
+    }
+
+    Ok(())
+}
+
+fn validate_message<T: ssz::Encode>(
+    public_key: &PublicKey,
+    message: &T,
```

Comment:
For simple types I'd probably inline. Curious if there was a special reason to use a generic type var here? (I'm fine with keeping it the way it is, just curious).

Reply:
Not any special reason really. I did struggle at other points sometimes when using Traits as parameters and had to add a bunch of `dyn` `impl` `Box` that I only partly understood. So I like the fact that I didn't have to do that here. 
But I'm not sure what the implications of these different approaches are beyond code readability.